gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java

```diff
@@ -138,16 +138,23 @@ public void apply(Project project) {
             task.setMainClassName(distributionExtension.getMainClass().get());
             task.setApplicationName(distributionExtension.getDistributionServiceName().get());
             task.setDefaultJvmOpts(distributionExtension.getDefaultJvmOpts().get());
+            task.dependsOn(manifestClassPathTask);
 
             JavaPluginConvention javaPlugin = project.getConvention().findPlugin(JavaPluginConvention.class);
-            task.setClasspath(jarTask.get().getOutputs().getFiles().plus(
-                    javaPlugin.getSourceSets().getByName("main").getRuntimeClasspath()));
+            if (distributionExtension.getEnableManifestClasspath().get()) {
+                task.setClasspath(manifestClassPathTask.get().getOutputs().getFiles());
+            } else {
+                task.setClasspath(
+                        jarTask.get().getOutputs().getFiles().plus(
```

Comment:
why did you replace the `main`s runtimeClasspath with the product dependencies configuration?

Reply:
copy-paste error